client/reader/reader.h

```diff
@@ -0,0 +1,25 @@
+#ifndef READER_H
+#define READER_H
+#include "../client.h"
+
+class Reader : private Client {
+private:
+  Reader(){};
```

Comment:
You made Reader a singleton that inherits from a regular class. Not the best design for sure ;)